torchvision/datasets/celeba.py

```diff
@@ -120,28 +120,32 @@ def download(self):
     def __getitem__(self, index):
         X = PIL.Image.open(os.path.join(self.root, self.base_folder, "img_align_celeba", self.filename[index]))
 
-        target = []
-        for t in self.target_type:
-            if t == "attr":
-                target.append(self.attr[index, :])
-            elif t == "identity":
-                target.append(self.identity[index, 0])
-            elif t == "bbox":
-                target.append(self.bbox[index, :])
-            elif t == "landmarks":
-                target.append(self.landmarks_align[index, :])
-            else:
-                # TODO: refactor with utils.verify_str_arg
-                raise ValueError("Target type \"{}\" is not recognized.".format(t))
-        target = tuple(target) if len(target) > 1 else target[0]
-
         if self.transform is not None:
             X = self.transform(X)
 
-        if self.target_transform is not None:
-            target = self.target_transform(target)
-
-        return X, target
+        if not self.target_type:
+            return X, None
+        else:
```

Comment:
I would maybe simplify the changes in this PR like the following, and only change the following places:
```python
if target:
    target = tuple(target) if len(target) > 1 else target[0]
    
    if self.target_transform is not None:
        target = self.target_transform(target)
else:
    target = None
```
Thoughts?

Reply:
I had doubts about keeping the diff small vs keeping the logic straightforward. Amended the commit following your advice.

Also, added a check for a meaningless combination of parameters (empty `target_type` with nonempty `target_transform`).